packages/mui-material/src/styles/createPalette.js

```diff
@@ -261,7 +265,7 @@ export default function createPalette(palette) {
     return color;
   };
 
-  const modes = { dark, light };
+  const modes = { dark: getDarkTokens(), light: getLightTokens() };
```

Comment:
I have to do this to guarantee that the result of `createTheme` is always a new object. It breaks my tests in `getTextDecoration.test.js`.

Reply:
But the result of `createTheme` is always a new object. What exactly broke without these changes? This looks off.

Reply:
Not for `palette.text.*`.

```js
const theme1 = createTheme();

theme1.palette.text.primary = "some-color"

const theme2 = createTheme();
console.log(theme2.palette.text.primary) // "some-color"
```

Reply:
Let's update the test to work with the current setup, and open a new PR for this and discuss there. Maybe we can just move the object definitions in the `createPalette` function.

Reply:
Reverted the change on `createPalette` and fixed the tests.